docs/examples/oceanmodelling.py

```diff
@@ -0,0 +1,509 @@
+# %%
+from jax.config import config
+
+config.update("jax_enable_x64", True)
+from dataclasses import dataclass
+
+from jax import hessian
+from jax.config import config
+import jax.numpy as jnp
+import jax.random as jr
+from jaxtyping import (
+    Array,
+    Float,
+    install_import_hook,
+)
+from matplotlib import rcParams
+import matplotlib.pyplot as plt
+import optax as ox
+import pandas as pd
+import tensorflow_probability as tfp
+
+with install_import_hook("gpjax", "beartype.beartype"):
+    import gpjax as gpx
+
+# Enable Float64 for more stable matrix inversions.
+key = jr.PRNGKey(123)
+plt.style.use(
+    "https://raw.githubusercontent.com/JaxGaussianProcesses/GPJax/main/docs/examples/gpjax.mplstyle"
+)
+colors = rcParams["axes.prop_cycle"].by_key()["color"]
+
+
+# %% [markdown]
+# ## Data Loading and problem setting
+# The real dataset has been binned into an $N=34\times16$ grid, equally spaced over the longitude-latitude interval $[-90.8,-83.8] \times [24.0,27.5]$. Each bin has a size $\approx 0.21\times0.21$, and contains the average velocity across all measurements that fall inside it.
+#
+# We will call this binned ocean data the ground truth, and label it with the vector field
+# $$
+# \mathbf{F} \equiv \mathbf{F}(\mathbf{x}),
+# $$
+# where $\mathbf{x} = (x^{(0)}$,$x^{(1)})^\text{T}$, with a vector basis in the standard Cartesian directions (dimensions will be indicated by superscripts).
+#
+# We shall label the ground truth $D_0=\left\{ \left(\mathbf{x}_{0,i} , \mathbf{y}_{0,i} \right)\right\}_{i=1}^N$, where $\mathbf{y}_i$ is the 2 dimensional velocity vector at the $i$th location, $\mathbf{x}_i$. The training dataset contains simulated measurements from ocean drifters $D_T=\left\{\left(\mathbf{x}_{T,i}, \mathbf{y}_{T,i} \right)\right\}_{i=1}^{N_T}$, $N_T = 20$ in this case (the subscripts indicate the ground truth and the simulated measurements respectively).
+#
+
+
+# %%
+# function to place data from csv into correct array shape
+def prepare_data(df):
+    pos = jnp.array([df["lon"], df["lat"]])
+    vel = jnp.array([df["ubar"], df["vbar"]])
+    # extract shape stored as 'metadata' in the test data
+    try:
+        shape = (int(df["shape"][1]), int(df["shape"][0]))  # shape = (34,16)
+        return pos, vel, shape
+    except KeyError:
+        return pos, vel
+
+
+# loading in data
+
+gulf_data_train = pd.read_csv(
+    "https://raw.githubusercontent.com/JaxGaussianProcesses/gpjaxstatic/main/data/gulfdata_train.csv"
+)
+gulf_data_test = pd.read_csv(
+    "https://raw.githubusercontent.com/JaxGaussianProcesses/gpjaxstatic/main/data/gulfdata_test.csv"
+)
```

Comment:
@ivanshalashilin - we have an issue with the links here. (Docs build fail and can't load this on my local machine).